plugins/ata/fu-plugin-ata.c

```diff
@@ -18,3 +18,11 @@ fu_plugin_init (FuPlugin *plugin)
 	fu_plugin_add_udev_subsystem (plugin, "block");
 	fu_plugin_set_device_gtype (plugin, FU_TYPE_ATA_DEVICE);
 }
+
+gboolean
+fu_plugin_device_created (FuPlugin *plugin, FuDevice *dev, GError **error)
+{
+	g_autofree gchar *tmp = fu_plugin_get_config_value (plugin, "UnknownOuiReport");
+	fu_ata_device_set_unknown_oui_report (FU_ATA_DEVICE (dev), g_strcmp0 (tmp, "true") == 0);
```

Comment:
I think you want to use `g_ascii_strcasecmp` no?

Reply:
Also I think you need to check that you don't get back `NULL` from `fu_plugin_get_config_value`, which might happen if `ata.conf` is destroyed or mangled.

Reply:
> g_ascii_strcasecmp

This isn't NULL safe.

> you need to check that you don't get back NULL

This is NULL safe: `g_strcmp0`

Maybe we want `fu_plugin_get_config_value_boolean()`?

Reply:
> fu_plugin_get_config_value_boolean

Yeah, can probably simplify some other uses elsewhere in the codebase too then.

Reply:
Something like https://github.com/fwupd/fwupd/pull/1791 ?

Reply:
Okay, fixed now.